torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,427 @@
+        """Verify the integrity of the dataset.
+
+        Args:
+            url: url to the file
+            md5: md5 of the file to be verified
+
+        Raises:
+            RuntimeError: if dataset is not found
+        """
+        modality_folder_name = url.split("/")[-1]
+        mod_fold_no_ext = modality_folder_name.split(".")[0]
+        modality_path = os.path.join(self.root, mod_fold_no_ext)
+        split_path = os.path.join(self.root, modality_folder_name)
+        if mod_fold_no_ext == "split_IDs":
+            modality_path = split_path
+
+        # Check if the files already exist
+        if os.path.exists(modality_path):
+            return
+
+        # Check if the zip files have already been downloaded, if so, extract
+        filepath = os.path.join(self.root, url.split("/")[-1])
+        if os.path.isfile(filepath) and filepath.endswith(".zip"):
+            if self.checksum and not check_integrity(filepath, md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            self._extract(url)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` directory and `download=False`, "  # noqa: E501
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+        elif url.endswith(".csv"):
+            # Download the split file
+            download_url(url, self.root, filename=os.path.split(url)[1], md5=None)
+
+        else:
+            # Download the dataset
+            self._download(url, md5)
+            self._extract(url)
+
+    def _download(self, url: str, md5: Optional[str]) -> None:
+        """Downloads a modality.
+
+        Args:
+            url: download url of a modality
+            md5: md5 of a modality
+        """
+        download_url(
+            url,
+            self.root,
+            filename=os.path.split(url)[1],
+            md5=md5 if self.checksum else None,
+        )
+
+    def _extract(self, path: str) -> None:
+        """Extracts a modality.
+
+        Args:
+            path: path to the modality folder
+        """
+        filepath = os.path.join(self.root, os.path.split(path)[1])
+        extract_archive(filepath)
+
+    def _merge_parts(self, modality: str) -> None:
+        """Merge the modalities that are downloaded and extracted in two parts.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the filename of the modality
+        """
+        # Create a new folder named after the 'modality' variable
+        modality_folder = os.path.join(self.root, modality)
+        os.makedirs(
+            modality_folder, exist_ok=True
+        )  # Will not raise an error if the folder already exists
```

Comment:
Can you put this comment on the line before so it doesn't get stretched to 3 lines?